### ISSUE TITLE
Problem controlling frame annotation in subplot

### ISSUE THREAD

**Issue by thatfaultguy** [GenericMappingTools/gmt#4235]
**Description of the problem**

I am using the inside MAP_FRAME_TYPE and would like change the frame annotation between subplots with the first subplot having annotated ticks and the following two subplots with just the tick marks. If I specify the -B option at the subplot begin line, then this seems to override the individual -B commands in the subplots.  

Furthermore, the ticks and annotations are behind the grd file (and only visible because it is transparent - I think), how can I bring them to the front? I don't want to use annotations outside the plot as it would create more white space between the subplot frames.

**Full script that generated the error**

link to data files used in the script:
https://drive.google.com/drive/folders/1SnAVMq4OleuOQMrr4jAg3BY4LcLZZbOF?usp=sharing

```
# PASTE CODE HERE
#!/bin/bash

cpt="roma0.cpt"
obs_file1="D118input.grd"
model_file1="D118_lmodel.grd"
residual_file1="D118_lresidual.grd"
fault="fault_1f_uniform_long.txt"


gmt begin insar_plot png
gmt set MAP_FRAME_TYPE inside 
    gmt subplot begin 3x3 -Fs3i -A 
        gmt subplot set 
            gmt grdimage $obs_file1 -R170/172/61.8/62.5 -C$cpt -t40 
            gmt plot $fault -W1p,black -BWeSn+t"Observed"
        gmt subplot set 
            gmt grdimage $model_file1 -R170/172/61.8/62.5 -C$cpt -t40 
            gmt plot $fault -Bwesn+t"Modelled" -W1p,black 
        gmt subplot set 
            gmt grdimage $residual_file1 -R170/172/61.8/62.5 -C$cpt -t40
            gmt plot $fault -Bwesn+t"Residual" -W1p,black
    gmt subplot end 
gmt end show 
```

**Full error message**

```
# PASTE ERROR MESSAGE HERE
```

**Actual outcome**

![insar_plot](https://user-images.githubusercontent.com/67931182/93647870-454d0780-f9be-11ea-8ebe-43d83a3fd3eb.png)

**Expected outcome**

I was hoping that only the first subplot would be annotated with the other subplots having only tickmarks. also the annotations should be completely opaque and at the 'top' layer

**System information**

* Operating system:macOS
* GMT version (6.1.1):

**Comment by joa-quim** [comment]
Hi, this is a question for the forum (not an issue that I can spot)

You are setting ``MAP_FRAME_TYPE inside`` before the subplot command so logically it applies to all panels. If you want it applied only to one of them, use it inside that particular ``subplot set``. Or use it in the command itself using the ``--MAP_FRAME_TYPE=inside`` mechanism.

**Comment by PaulWessel** [comment]
We may have an issue if **MAP_FRAME_TYPE** is set to inside before **subplot begin** is called then that should affect the math on the margin between panels.  Right now we are computing the space needed for annotations but they are obviously on the inside and don't clog the margins.

**Comment by thatfaultguy** [comment]
Sorry if I put this in the wrong place, I would remove/resolve it but I'm not sure how to do that.

**Comment by thatfaultguy** [comment]
I think is Paul is right, I tried the -M option to get the frames closer together and the only time it really worked well was when I specified -B+n 

The other potential issue is that the frame marks and annotations are drawn as the first layer, and thus could be completely obscured if the grid file is opaque

**Comment by PaulWessel** [comment]
Yes, we also start off each new panel with the implicit **basemap** call, but clearly in the case of inside annotations we need to split that into a blank frame up front (to initialize the new panel) and then complete the panel with the intended **basemap** when you move to the next panel or you are done.

**Comment by PaulWessel** [comment]
Some further clarification (mostly for me) as to how this might work.  When **MAP_FRAME_TYPE**=_inside_ is in effect, then all axis annotation(including optional labels) are placed inside.  But plot title is still outside of course.  E.g.,

![map](https://user-images.githubusercontent.com/26473567/93822678-466b7800-fbfc-11ea-9290-1f6f0cc6091e.png)

1. So if inside is set prior to **gmt subplot begin** then all the fluff calculations we do to make space for annotations between panels should be set to zero.  This is straight forward.  Because this is a grid layout, should anyone switch to inside for specific panels then that will not affect the initial fluff and margin calculations.
2. The trickier think is to check that if inside is set the ticks and annotations must happen at the end of the panel. The way it works now is that the first command in a panel must be a plotting command and we will add default **-B** settings to that command.  Ideally, we would change this if inside is set so that we add it to the last plot command, or add a special **basemap** call at the end.  Perhaps the best solution is to always end with a separate **basemap** call since that would be the same for inside and outside.  We have checks when we just finish a panel and complete things like legends so it shouldn't be too hard to add that overlay at the end.

I think this is my plan for addressing this problem.
